lib/miq_ldap.rb

```diff
@@ -266,16 +267,30 @@ def normalize(dn)
     dn.split(",").collect { |i| i.downcase.strip }.join(",")
   end
 
+  def self.dn?(str)
+    Net::LDAP::DN.new(str).to_a.any?
```

Comment:
Off PR, we chatted about this PR and @Fryguy answered my questions.  Basically, this change of `dn?` changes from an inefficient regex to a statemachine to parse the dn in [Net::LDAP](https://github.com/ruby-ldap/ruby-net-ldap/blob/master/lib/net/ldap/dn.rb#L40-L176).  This is what resolves the CWE issues.